### PR TITLE
Parameters defined in order to create basic shapes

### PR DESCRIPTION
**Pull request by jaimeatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
The parameters needed for each shape has been defined. Currently the basic shapes can be positioned and scaled. The basic shapes are not orientable in this stage.

### CODE REVIEW

File: COMETwebapp/Scene.cs [hunk 1]
```diff
@@ -30,16 +30,59 @@ namespace COMETwebapp
...
+
+        /// <summary>
+        /// Width parameter short name
+        /// </summary>
+        public const string WidthShortName = "wid_diameter";
+
+        /// <summary>
+        /// Diameter parameter short name
+        /// </summary>
+        public const string DiameterShortName = WidthShortName;
```
Comment (alexatrhea) [inline COMETwebapp/Scene.cs:69]: is this correct?

File: cls [hunk 2]
```diff
@@ -0,0 +1,148 @@
+* 47b85ae Defined the parameters and add some tests
```
Comment (alexatrhea) [inline cls:1]: please remove this file

**Review by alexatrhea — CHANGES_REQUESTED** [review]

File: COMETwebapp/Primitives/BasicPrimitive.cs [hunk 3]
```diff
@@ -133,48 +142,81 @@ public void ResetRotation()
...
-                parameterBase = elementUsage.ElementDefinition.Parameter.FirstOrDefault(x => x.ParameterType.ShortName == shortName
-                          && x.ParameterType is CompoundParameterType);
+                parameterBase = elementUsage.ElementDefinition.Parameter.FirstOrDefault(x => x.ParameterType.ShortName == parameterTypeShortName
+                                                                                             && x.ParameterType.GetType() == Scene.GetParameterTypeFromParameterShortName(parameterTypeShortName));
             }
 
             if (parameterBase is not null)
             {
-                foreach (var actualFiniteState in states)
+                if(states.Count > 0)
```
Comment (samatrhea) [inline COMETwebapp/Primitives/BasicPrimitive.cs:193]: the `Count` is not required, of the list is empty the for each will not hit, please remove

File: COMETwebapp/Primitives/Cone.cs [hunk 4]
```diff
@@ -71,5 +75,29 @@ public Cone(double x, double y, double z, double radius, double height)
...
             this.Height = height;
         }
+
+        /// <summary>
+        /// Set the dimensions of the <see cref="BasicPrimitive"/> from the <see cref="ElementUsage"/> parameters
+        /// </summary>
+        /// <param name="elementUsage">the <see cref="ElementUsage"/> used for the dimensioning</param>
+        /// <param name="selectedOption">the current <see cref="Option"/> selected</param>
+        /// <param name="states">the <see cref="ActualFiniteState"/> that are going to be used to dimensioning the <see cref="BasicPrimitive"/></param>
+        public override Task SetDimensionsFromElementUsageParameters(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states)
```
Comment (samatrhea) [inline COMETwebapp/Primitives/Cone.cs:85]: if none of the logic requires this method to be a `Task`, please refactor it to return `void`

File: COMETwebapp/Scene.cs (hunk 1 above)
Comment (jaimeatrhea) [reply COMETwebapp/Scene.cs:69]: Hmm it is correct. The diameter and width share the shortName but in terms of calling the field I thought that I would be more clear if the names are different.

File: COMETwebapp/Primitives/Cube.cs [hunk 5]
```diff
@@ -80,5 +84,40 @@ public Cube(double x, double y , double z, double width, double height, double d
...
             this.Depth = depth;
         }
+
+        /// <summary>
+        /// Set the dimensions of the <see cref="BasicPrimitive"/> from the <see cref="ElementUsage"/> parameters
+        /// </summary>
+        /// <param name="elementUsage">the <see cref="ElementUsage"/> used for the dimensioning</param>
+        /// <param name="selectedOption">the current <see cref="Option"/> selected</param>
+        /// <param name="states">the <see cref="ActualFiniteState"/> that are going to be used to dimensioning the <see cref="BasicPrimitive"/></param>
+        public override Task SetDimensionsFromElementUsageParameters(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states)
```
Comment (samatrhea) [inline COMETwebapp/Primitives/Cube.cs:94]: see previous comment regarding the use of `Task`

File: COMETwebapp/Primitives/IShapeFactory.cs [hunk 6]
```diff
@@ -39,6 +39,6 @@ public interface IShapeFactory
         /// <param name="states">The list of <see cref="ActualFiniteState"/> that are active</param>
         /// <param name="basicShape">The basic shape of type <see cref="Primitive"/></param>
         /// <returns></returns>
-        bool TryGetPrimitiveFromElementUsageParameter(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states, out Primitive basicShape);
+        Task<Primitive> TryGetPrimitiveFromElementUsageParameter(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states);
```
Comment (samatrhea) [inline COMETwebapp/Primitives/IShapeFactory.cs:42]: see previous comment regarding the use of `Task`

File: COMETwebapp/Primitives/ShapeFactory.cs [hunk 7]
```diff
@@ -40,9 +40,8 @@ public class ShapeFactory : IShapeFactory
         /// <param name="elementUsage">The <see cref="ElementUsage"/> used for creating a <see cref="Primitive"/></param>
         /// <param name="selectedOption">The current <see cref="Option"/> selected</param>
         /// <param name="states">The list of <see cref="ActualFiniteState"/> that are active</param>
-        /// <param name="basicShape">The basic shape of type <see cref="Primitive"/></param>
-        /// <returns></returns>
-        public bool TryGetPrimitiveFromElementUsageParameter(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states, out Primitive basicShape)
+        /// <returns>A <see cref="Task" of <see cref="Primitive"/> result/></returns>
+        public Task<Primitive> TryGetPrimitiveFromElementUsageParameter(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states)
```
Comment (samatrhea) [inline COMETwebapp/Primitives/ShapeFactory.cs:44]: see previous comment regarding the use of `Task`

File: COMETwebapp/Primitives/Sphere.cs [hunk 8]
```diff
@@ -62,5 +66,23 @@ public Sphere(double x, double y, double z, double radius)
...
             this.Radius = radius;
         }
+
+        /// <summary>
+        /// Set the dimensions of the <see cref="BasicPrimitive"/> from the <see cref="ElementUsage"/> parameters
+        /// </summary>
+        /// <param name="elementUsage">the <see cref="ElementUsage"/> used for the dimensioning</param>
+        /// <param name="selectedOption">the current <see cref="Option"/> selected</param>
+        /// <param name="states">the <see cref="ActualFiniteState"/> that are going to be used to dimensioning the <see cref="BasicPrimitive"/></param>
+        public override Task SetDimensionsFromElementUsageParameters(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states)
```
Comment (samatrhea) [inline COMETwebapp/Primitives/Sphere.cs:76]: see previous comment regarding the use of `Task`

File: COMETwebapp/Primitives/Torus.cs [hunk 9]
```diff
@@ -71,5 +75,29 @@ public Torus(double x, double y, double z, double diameter, double thickness)
...
             this.Thickness = thickness;
         }
+
+        /// <summary>
+        /// Set the dimensions of the <see cref="BasicPrimitive"/> from the <see cref="ElementUsage"/> parameters
+        /// </summary>
+        /// <param name="elementUsage">the <see cref="ElementUsage"/> used for the dimensioning</param>
+        /// <param name="selectedOption">the current <see cref="Option"/> selected</param>
+        /// <param name="states">the <see cref="ActualFiniteState"/> that are going to be used to dimensioning the <see cref="BasicPrimitive"/></param>
+        public override Task SetDimensionsFromElementUsageParameters(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states)
```
Comment (samatrhea) [inline COMETwebapp/Primitives/Torus.cs:85]: see previous comment regarding the use of `Task`

File: cls (hunk 2 above)
Comment (samatrhea) [inline cls]: i doubt this file needs to be commited, please remove

**Review by samatrhea — CHANGES_REQUESTED** [review]

**Review by samatrhea — APPROVED** [review]

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**